includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1111,6 +1111,8 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 			// phpcs:ignore WordPress.Security.NonceVerification.Missing
 			$platform_checkout_intent_id = sanitize_user( wp_unslash( $_POST['platform-checkout-intent'] ?? '' ), true );
 
+			$statement_descriptor = $this->get_statement_descriptor( $order->get_payment_method(), $order );
```

Comment:
Now that you confirmed that the full statement descriptor comes directly from Stripe, perhaps we don't want to pass it to the payment intent, unless it needs to be changed.

The full statement descriptor should always be the same from the Stripe dashboard, correct?`https://dashboard.stripe.com/test/connect/accounts/{acct_id}/identity`

I suppose what differs from the Stripe plugin now is that we don't (or can't), set the full statement descriptor for the standard Stripe account directly, so that's why we allow a custom statement descriptor from the plugin settings (I didn't check this assumption).

Since we can set the express account statement descriptor in WCPay, we may not want to pass it along with the request, since it will always default to its full form anyway. We also may not want to do that to not risk providing an invalid statement descriptor somehow.

Now that I look at the Stripe docs more carefully, we may actually want to save the short statement descriptor in the account instead: https://stripe.com/docs/api/accounts/object#account_object-settings-card_payments-statement_descriptor_prefix

And then pass a [suffix](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-statement_descriptor_suffix).

So in the end, this could be refactored like so:

```php
$statement_descriptor_suffix = $this->maybe_use_statement_descriptor_suffix( $order->get_payment_method(), $order );
```

```php
public function maybe_use_statement_descriptor_suffix( $payment_method, $order ) {
    $statement_descriptor_prefix           = $this->get_account_statement_descriptor_prefix();
    $is_short_statement_descriptor_enabled = 'yes' === $this->get_option( 'is_short_statement_descriptor_enabled' );

    if ( in_array( $payment_method, [ 'card', 'woocommerce_payments' ], true ) && $is_short_statement_descriptor_enabled && ! empty( $statement_descriptor_prefix ) ) {
        // TODO: Refactor this:
        return WC_Payments_Utils::get_dynamic_statement_descriptor( $order );
    }

    return null;
}
```

`get_account_statement_descriptor_prefix` would grab the [statement descriptor prefix](https://stripe.com/docs/api/accounts/object#account_object-settings-card_payments-statement_descriptor_prefix) from the account, similar to `get_account_statement_descriptor`, but probably without logging errors.

Then this function should only return the `* #123` part, which is the suffix.

And then we need to rename the additional parameter `statement_descriptor` to `statement_descriptor_suffix`.

Note: I didn't test the code above.

Reply:
>perhaps we don't want to pass it to the payment intent, unless it needs to be changed.

Agreed. I tweaked the logic [here](https://github.com/Automattic/woocommerce-payments/pull/3643/files#diff-bcfd303d507c33a27671935e724ee64d0df30ff2a18bb6ef4c94f28d48190124R2369-R2371) to only return it when payment method is not a card, which doesn't set a statement descriptor even if it's set on Stripe – that's why I also recommended testing a non-card payment method, such as SEPA.

>we may actually want to save the short statement descriptor in the account instead
I've tried that approach and it didn't play well with the regular and suffix statement descriptors, not sure why. I've tried to follow [Stripe docs](https://stripe.com/docs/account/statement-descriptors#dynamic) and sent different combinations as well as setting all of them together and the payment intent always ended up having the full statement descriptor set in it.

Reply:
@cesarcosta99 I did some testing and was able to set the short statement descriptor (aka `statement_descriptor_prefix`) in the Stripe account level and create a PaymentIntent using that.

1. Changed the server to also accept a "statement_descriptor_suffix" as an argument [here](https://github.com/Automattic/woocommerce-payments-server/blob/9c8fdef554a456ed25a8addbabe5f41cd10e77b5/server/wp-content/rest-api-plugins/endpoints/wcpay/class-intentions-controller.php#L151) and [here](https://github.com/Automattic/woocommerce-payments-server/blob/9c8fdef554a456ed25a8addbabe5f41cd10e77b5/server/wp-content/rest-api-plugins/endpoints/wcpay/class-intentions-controller.php#L234).
2. Manually updated my account using `curl` to have a card statement descriptor prefix:
```bash
curl https://api.stripe.com/v1/accounts/{WCPay_Dev_CONNECTED_ACCOUNT_ID} \
  -u "{WCPay_Dev_SECRET_KEY}:" \ # Remember the ":" at the end of the secret key
  -d "settings[card_payments][statement_descriptor_prefix]"="WOO.COM"
```
3. Add the `statement_descriptor_suffix` to the WCPay server intentions request to something like "ID: 1234". (Like what you did with `$request->set_statement_descriptor()`.

The only issue I encountered was that I needed to put a Latin character on the statement descriptor suffix, otherwise Stripe would throw an error:
> invalid_request_error - statement_descriptor_suffix
> The statement descriptor must contain at least one Latin character.